File: _practice/1038-Binary-Search-Tree-to-Greater-Sum-Tree/iterative-reverse-inorder.py
class Solution:
    def bstToGst(self, root: TreeNode) -> TreeNode:
        node_sum = 0
        stack = []
        node = root

        while stack or node is not None:
            while node is not None:
                stack.append(node)
                node = node.right

            node = stack.pop()     
            node_sum += node.val
            node.val = node_sum

            node = node.left

        return root

File: _practice/1038-Binary-Search-Tree-to-Greater-Sum-Tree/iterative-reverse-inorder.js
var bstToGst = function (root) {
  const stack = [];
  let nodes_sum = 0;
  let node = root;

  while (node || stack.length > 0) {
    while (node) {
      stack.push(node);
      node = node.right;
    }

    node = stack.pop();
    nodes_sum += node.val;
    node.val = nodes_sum;

    node = node.left;
  }

  return root;
};
